javascript/javascript1/week2/Homework.js

```diff
@@ -0,0 +1,146 @@
+        if(gender === "male") {
+        return "Lord "+`${firstname} ${surname}`;
+    } else if (gender === "female"){
+      return "Lady " + `${firstname} ${surname}`;
+    } 
+  } else {
+      return `${firstname} ${surname}`;
+    }
+  }; 
+
+    console.log(getFullname("Benjamin", "Hughes","male",true));
+    console.log(getFullname("Oana", "Loredana","female", true));
+    console.log(getFullname(" "," ","female",true));
+
+    // Event application
+
+function getEventWeekday (daysFromToday) {
+  const weekdays = ["Monday","Tuesday" , "Wednesday" , "Thursday" , "Friday" , "Saturday" , "Sunday"];
+  const today = weekdays.indexOf("Sunday");  //I can choose any day from the list
+  const eventDay =  (today + daysFromToday) % 7;
+  return weekdays[eventDay];
+};
+console.log(getEventWeekday(5)); // Logs out "Friday"
+
+
+    //*********** */ Weather wear********
+
+    function howIsWeather(temperature) {
+      if (temperature >= 30 ) {
+         return `The temperature is ${temperature} degrees, we recommand to wear shorts and t-shirt.`;
+    }else if (temperature >= 15 && temperature < 15) {
+         return  `The temperature is ${temperature} degrees, we recommand to wear pants and blouse.`;
+    }else {
+         return `The temperature is ${temperature} degrees ,you should wear jacket and gloves.`;
+    }
+  }
+
+    let getRandomDegrees = Math.floor(Math.random() * (40 - (-10) + 1)) + (-10); //return a random nr between -10 and 40.
+
+    const clothesToWear = howIsWeather(getRandomDegrees)
+    console.log(clothesToWear);
+
+
+                // ********** Student manager*************
+
+  const class07Students = [];
+  function addStudentToClass(studentName) {
+  
+    // Check if the class has more then 6 students.
+    if (class07Students.length >= 6 && studentName !== "Queen"){
+        console.log('Cannot add more students to class 07');
+        return;
+}
+      // Check if the student is already in the class.
+    if(class07Students.includes(studentName)) {
+      console.log(`Student ${studentName} is already in class`);
+      return;
+    }
+
+    // If the Queen is being added, always add her to the class
+    if (studentName === 'Queen'){
+       class07Students.push(studentName);
+       return;
+     }
+
+     // Check if the student name is not an empty string.
+     if(studentName.trim() === ""){
+      console.log('You cannot add an empty string to a class')
+      return;
+     }
+     class07Students.push(studentName);
+     return;
+    }
+     // Return the number of students in the class
+  function getNumberOfStudents() {
+  return class07Students.length;
+}
+console.log("Number of students:", getNumberOfStudents());
+
+
+addStudentToClass("Emma");
+addStudentToClass("David");
+addStudentToClass("Eva");
+addStudentToClass("Maria");
+addStudentToClass("Vasile");
+addStudentToClass("Nicoleta");
+addStudentToClass("Diana");
+addStudentToClass("Oana");
+addStudentToClass("Queen");
+addStudentToClass("Queen");
```

Comment:
Very well testing again for "Queen". You can also add a test for the case with empty string.

Reply:
I have tested and always the number of students from the class  is always 0 and i don't know why.